examples/travis/check-results.py

```diff
@@ -47,15 +50,25 @@ def test_power(output, ref, ref_z):
                     abs(ref.power['power'])[2:8] * D ** 2,
                         rtol=0.01, atol=0.0)
 
+def invwindow(kk):
+    """Sunc function, inverse window of the CIC."""
+    #2 pi nmesh / box
+    grid = 2* math.pi * 128 / 512.
+    return 1./numpy.sinc(kk/grid)**6
 
-# asserting the initial power spectrum is 5% accurate
+# asserting the initial power spectrum is 1% accurate
 print("testing IC power")
-pklin = LinearPower(Planck15, redshift=ref_z)
+pklin = LinearPower(WMAP9, redshift=ref_z)
+#This checks that the power spectrum loading and rescaling code is working.
+genpk = numpy.loadtxt("output/inputspec_IC.txt")
+assert_allclose(pklin(genpk[:,0]), genpk[:,1], rtol=5e-4, atol=0.0)
+#This checks that the output is working
 print(ref_z, ref.power['power'][2:8])
 print(pklin(ref.power['k'][2:8]))
-assert_allclose(abs(ref.power['power'])[2:8],
+#Deconvolve by the CIC window. This is not exact.
```

Comment:
There is no need to deconvolve CIC for the measurements -- nbodykit does its own corrections. 
Were you correcting for the CIC in ZA?

Reply:
Possibly. I was just following what GenPK does, which I worked it out many years ago and have since forgotten. It uses sinc^{-2} for each k dimension, so sinc^{-6} in total, applied to delta. So for P(k) it should be sinc^{-12}. Looks like one factor of sinc is missing here.

Reply:
I looked through the nbodykit code but got a little lost in the python - it looks like the right factor of sinc is applied, but in real space before the fourier transform, which I don't understand fully. Since the factor applied is the same I also do not understand why it differs from GenPK unless there is a bug somewhere.

Reply:
I think you just need to look at this figure: http://nbodykit.readthedocs.io/en/latest/cookbook/interlacing.html ; nbodykit uses is interlacing and TSC -- you can think of the nbodykit power spectrum estimator as 'unbiased' at all scales.


The CIC window in GenIC does 'smooth-out' ZA; but since the displacment invokes non-linear coupling, I don't see why you'll need a correction -- did your correction improve agreement against linear power?  ZA is not even supposed to be the same as the linear power -- at k~0.05 it is already 2% difference even in the limit where the CIC window is very small.

Reply:
There might be a factor of 2 in your sinc kernel -- did you check your GenPK give a flat power back with a uniform field? The SINC correction is supposed to give you a flat power on shot-noise like input, see here:
[mozilla.pdf](https://github.com/rainwoodman/MP-Gadget/files/1406083/mozilla.pdf)

You can also see that the TSC+interlacing gives a slight up-turn for pure shotnoise, because it is supposed work better on LCDM-like power(not dominated by shotnoise at high-k). The default (CIC+compensation) in nbodykit is a good estimator when Nmesh is so great that the field is already shot-noise. I somehow misremembered that the default is the TSC+interlacing setup.

Reply:
Also the compensation is applied in fourier space, as declared here: https://github.com/bccp/nbodykit/blob/d554caf8b6db15b1d7e71fa9a684dae7834e814b/nbodykit/base/catalogmesh.py#L502

Reply:
Right - so it looks like NBodykit and GenPK are doing the same thing. I double-checked that nbodykit gives exactly the same answer with tsc once the window function is corrected for (impressively so!). The window function in the linked document is the one implemented in GenPK. GenPK also gives a totally flat spectrum for white noise down to the Nyquist frequency of the particles. Also *both* codes give results that are invariant to the grid size. So then I wonder why they differ from each other.